pkg/driver/kubernetes/run.go

```diff
@@ -187,90 +195,50 @@ func (k *kubernetesDriver) runContainer(
-		pod.Spec.Containers[0].VolumeMounts = append(volumeMounts, podTemplate.Spec.Containers[0].VolumeMounts...)
-		pod.Spec.Containers[0].SecurityContext = &corev1.SecurityContext{
+	pod.ObjectMeta.Name = id
+	pod.ObjectMeta.Labels = labels
+
+	pod.Spec.ServiceAccountName = serviceAccount
+	pod.Spec.InitContainers = append(initContainer, pod.Spec.InitContainers...)
+	devPodContainer := corev1.Container{
+		Name:      "devpod",
+		Image:     imageName,
+		Command:   []string{entrypoint},
+		Args:      args,
+		Env:       envVars,
+		Resources: parseResources(k.config.Resources, k.Log),
+		SecurityContext: &corev1.SecurityContext{
 			Capabilities: capabilities,
 			Privileged:   mergedConfig.Privileged,
 			RunAsUser:    &[]int64{0}[0],
 			RunAsGroup:   &[]int64{0}[0],
 			RunAsNonRoot: &[]bool{false}[0],
-		}
-		if k.config.Resources != "" {
-			pod.Spec.Containers[0].Resources = parseResources(k.config.Resources, k.Log)
-		}
-		pod.Spec.RestartPolicy = corev1.RestartPolicyNever
-		pod.Spec.Volumes = append(
-			[]corev1.Volume{
-				{
-					Name: "devpod",
-					VolumeSource: corev1.VolumeSource{
-						PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
-							ClaimName: id,
-						},
-					},
-				},
-			},
-			podTemplate.Spec.Volumes...,
-		)
-	} else {
-		pod = &corev1.Pod{
-			TypeMeta: metav1.TypeMeta{
-				Kind:       "Pod",
-				APIVersion: corev1.SchemeGroupVersion.String(),
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:   id,
-				Labels: labels,
-			},
-			Spec: corev1.PodSpec{
-				ServiceAccountName: serviceAccount,
-				InitContainers:     initContainer,
-				Containers: []corev1.Container{
-					{
-						Name:         "devpod",
-						Image:        imageName,
-						Command:      []string{entrypoint},
-						Resources:    parseResources(k.config.Resources, k.Log),
-						Args:         args,
-						Env:          envVars,
-						VolumeMounts: volumeMounts,
-						SecurityContext: &corev1.SecurityContext{
-							Capabilities: capabilities,
-							Privileged:   mergedConfig.Privileged,
-							RunAsUser:    &[]int64{0}[0],
-							RunAsGroup:   &[]int64{0}[0],
-							RunAsNonRoot: &[]bool{false}[0],
-						},
-					},
-				},
-				RestartPolicy: corev1.RestartPolicyNever,
-				Volumes: []corev1.Volume{
-					{
-						Name: "devpod",
-						VolumeSource: corev1.VolumeSource{
-							PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
-								ClaimName: id,
-							},
-						},
-					},
+		},
+	}
+	if pod.Spec.Containers != nil && len(pod.Spec.Containers) > 0 && pod.Spec.Containers[0].Env != nil {
+		devPodContainer.Env = append(devPodContainer.Env, pod.Spec.Containers[0].Env...)
+	}
+	if pod.Spec.Containers != nil && len(pod.Spec.Containers) > 0 && pod.Spec.Containers[0].VolumeMounts != nil {
+		devPodContainer.VolumeMounts = append(devPodContainer.VolumeMounts, pod.Spec.Containers[0].VolumeMounts...)
+	}
+
+	pod.Spec.Containers = append(pod.Spec.Containers, devPodContainer)
```

Comment:
This looks like a problem to me, we add a new container to already existing containers? And before we join env and volume mounts from this container to the new container? Whats the goal here?

Reply:
yeah it's kinda weird but the goal is to preserver existing containers you want to be schedule on this pod together with the devpod container. 
Also not sure if we should do this kind of merging with the podd manifest at all for the containers (and rather just merge pod specific configuration + initContainers) tbh but that's what we had in the other PR